docs/src/user-manual/components.md

```diff
@@ -1,30 +1,141 @@
+reliable communication between the ingest application and the processing storage
+service.
+
+## Workflow engine
+
+The Workflow engine is responsible for orchestrating Enduro's workflows - that
+is, for kicking off preservation actions and other tasks, managing them, and
+recording them as auditable events. It also manages retries and timeouts,
+resulting in a reliable platform that can process digital objects for
+preservation in a highly automated environment.
+
+Currently, Enduro Ingest uses [Temporal](https://temporal.io/) as its workflow
+engine.
+
+## Application code
+
+The application code is where the business logic guiding Enduro's workflows is
+written. It is the glue that helps each component in the Enduro Ingest stack act
+in concert, to assist operators achieve their preservation goals.
+
+Enduro  uses the Go programming language ([GoLang](https://go.dev/)) for its
+application code.
+
+## Application Programming Interface (API)
+
+The [API](https://en.wikipedia.org/wiki/API) is how Enduro Ingest communicates
+with its [user interface](#user-interface), and how external users and
+applications can interact with Enduro. The Enduro Ingest API defines multiple
+services which map to resources in [REST](https://en.wikipedia.org/wiki/REST).
+Services define their own methods, errors, and so on.
+
+Enduro's application code uses the [Goa design language](https://goa.design/) to
+structure its API design. The API itself uses [Swagger](https://swagger.io/) to
+generate its documentation and ensure it conforms to
+[OpenAPI](https://www.openapis.org/) design specifications.
+
+## Identity & access management (IAM)
+
+This component provides authentication as well as role-based access control
+([RBAC](https://en.wikipedia.org/wiki/Role-based_access_control)) to manage user
+permissions and roles in Enduro. [API](#application-programming-interface-api)
+requests must be authorized by the IAM component before their contents are
+delivered or displayed in the [user interface](#user-interface), ensuring that
+sensitive information is only shared with authorized users and/or applications.
+
+[Keycloak](https://www.keycloak.org/) is currently used for Enduro's IAM
+component. Keycloak is based on standard protocols and provides support for
+OpenID Connect, OAuth 2.0, and SAML.
+
+## User interface
+
+Enduro's web-based user interface is the graphical method by which operators can
+interact with Enduro to see and manage ingest and preservation workflows, view
+information about packages, and more. Key information from other components and
+applications (such as tasks executed by the [preservation
+engine](#preservation-engine) in a workflow) is also collected via API and shown
+in the Enduro Ingest user interface when relevant.
+
+Enduro's user interface is built using [Vue.js](https://vuejs.org/) and
+[Bootstrap](https://getbootstrap.com/).
+
+## Preservation engine
+
+This component manages the transformation of Processing Information Packages
```

Comment:
Again, I'd prefer the component name to start the section rather than "This component...".

Reply:
:+1: